### ISSUE TITLE
401 Unauthorized when accessing account on Ubuntu 20.04 with ssh

### ISSUE THREAD

**Issue by nschmeller** [ankitpokhrel/jira-cli#133]
Hi! I love this project, it's the solution I never knew I needed. 

The CLI isn't working for me, and I suspect because it's because my company account only allows us to sign in with Okta. 

Steps I followed:
* Installed the binary on my dev environment using the `go get` option
* Created a JIRA API key and set it as an environment variable
* Ran `jira init`
  * Typed in the URL of my company's (cloud) JIRA server
  * Typed in my login email

After I hit enter on my login email, I got this output:
```
⠋ Verifying login details...
✗ Received unexpected response '401 Unauthorized' from jira. Please try again.
```

I'm assuming that this problem is happening because I use Okta to login to Jira. Is there anything I can do to get around this? Thanks!

**Comment by ankitpokhrel** [comment]
Hi @nschmeller, 

Glad to know that you like the project :)

Did you set the token `JIRA_API_TOKEN` (and is it available in the shell?) before running `jira init`? You should be able to get the token from [this link](https://id.atlassian.com/manage-profile/security/api-tokens).

**Comment by nschmeller** [comment]
I think I did, yes! 

I added it to my .zshrc and `source`d it. I get the following output:
```
 ~ echo $JIRA_API_TOKEN
XXXXXXXXXXXXXXXXXXXX
```
where the Xs are the token I remember putting in there.

**Comment by ankitpokhrel** [comment]
Can you check if you are able to get response from Jira using your token?

```sh
curl --request GET \
  --url 'https://<your-server-host>/rest/api/3/myself' \
  --user "<your-email>:$JIRA_API_TOKEN"
```

Also to double check: is this a on-premise Jira installation or a Jira cloud?

**Comment by nschmeller** [comment]
Yup, I get a response that has a bunch of correct, nontrivial info in it!

This is Jira cloud.

**Comment by ankitpokhrel** [comment]
That's strange because the tool calls same endpoint to verify the token. Are you sure your host url and email is correctly configured with no accidental special character or spaces? 🤔

**Comment by nschmeller** [comment]
Very strange! I triple checked that I was typing in the server and email correctly, in the form `https://<company name>.atlassian.net` for server. I also double checked that the email I typed in matched what I see at https://id.atlassian.com/manage-profile/email. 

I think it's some issue with my platform or setup that's doing weird things to text. This is on an Ubuntu 20.04 instance that I'm sshing into. I just tried installing on my Mac with Homebrew, and it's working perfectly there.

**Comment by infomiho** [comment]
I've added the JIRA_API_TOKEN to my .zshrc on my MacOS device but it didn't work.
I've added it like this
```
JIRA_API_TOKEN=XXXXXXXXXXXXXXX
```
Then I got the same error as the original issue. But it worked with cURL.

But worked in the end was adding it like this:
```
export JIRA_API_TOKEN=XXXXXXXXXXXXXXX
```

**Comment by joelyjoel** [comment]
I'm getting a similar issue. When I do

```
curl --request GET \
  --url 'https://<your-server-host>/rest/api/3/myself' \
  --user "<your-email>:$JIRA_API_TOKEN" -I
```

with my email and host I get:
```
HTTP/2 401 
server: AtlassianProxy/1.19.3.1
content-type: text/plain
strict-transport-security: max-age=315360000; includeSubDomains; preload
date: Tue, 30 Nov 2021 13:42:58 GMT
atl-traceid: b3054b2cf459f8ac
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
expect-ct: report-uri="https://web-security-reports.services.atlassian.com/expect-ct-report/global-proxy", enforce, max-age=86400
```

**Comment by joelyjoel** [comment]
Update: This issue was resolved when I switched my shell from fish to zsh

**Comment by seag-rvc** [comment]
I had a similar issue while using Ubuntu 18.04 in WSL2. For some reason adding `JIRA_API_TOKEN` in `$HOME/.bashrc` and then `source $HOME/.bashrc` did not work. Only after issuing `export JIRA_API_TOKEN=XXXXXX` did.

**Comment by NumberVII** [comment]
> I'm getting a similar issue. When I do
> 
> ```
> curl --request GET \
>   --url 'https://<your-server-host>/rest/api/3/myself' \
>   --user "<your-email>:$JIRA_API_TOKEN" -I
> ```
> 
> with my email and host I get:
> 
> ```
> HTTP/2 401 
> server: AtlassianProxy/1.19.3.1
> content-type: text/plain
> strict-transport-security: max-age=315360000; includeSubDomains; preload
> date: Tue, 30 Nov 2021 13:42:58 GMT
> atl-traceid: b3054b2cf459f8ac
> x-xss-protection: 1; mode=block
> x-content-type-options: nosniff
> expect-ct: report-uri="https://web-security-reports.services.atlassian.com/expect-ct-report/global-proxy", enforce, max-age=86400
> ```
> 
> My company uses google for Jira sign in. When I follow the `jira init` flow:
> 
> ```
> ⠼ Verifying login details... 
> ✗ Received unexpected response '401 Unauthorized' from jira. Please try again.
> ```
> 
> I suspect this 401 may be for the same reason

My company uses Microsoft SSO, got same 401 with curl and jira init. Are you able to curl and jira init after switching shell?

**Comment by dofinn** [comment]
```
[~/g/s/g/d/jira-cli]─[main]── ─ (⎈ XXX) rg 'JIRA_AUTH'
README.md
120:     a `JIRA_API_TOKEN` variable. In addition to this, set `JIRA_AUTH_TYPE` env to `bearer`.
137:default. If you want to use PAT, you need to set `JIRA_AUTH_TYPE` as `bearer`.
```

Additionally, the `JIRA_AUTH_TYPE` var does nothing? debugging the auth to local server with this var set as "bearer" as described in the readme still uses basic and fails... exact same token works using go-jira

**Comment by ankitpokhrel** [comment]
Closing as the original issue is fixed with the [latest release](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.0.0). @NumberVII please open a new issue with required details, your issue is something different.
